docs/docs/configuration/overview.md

```diff
@@ -354,6 +354,71 @@ It is recommended to use `--session-store-type=redis` when expecting large sessi
 
 You have to substitute *name* with the actual cookie name you configured via --cookie-name parameter. If you don't set a custom cookie name the variable  should be "$upstream_cookie__oauth2_proxy_1" instead of "$upstream_cookie_name_1" and the new cookie-name should be "_oauth2_proxy_1=" instead of "name_1=".
 
+## Configuring for use with the Traefik (v2) `ForwardAuth` middleware
+
+The [Traefik v2 `ForwardAuth` middleware](https://doc.traefik.io/traefik/middlewares/forwardauth/) allows Traefik to authenticate requests via the oauth2-proxy's `/oauth2/auth` endpoint, which only returns a 202 Accepted response or a 401 Unauthorized response without proxying the request through. For example, on Dynamic File (YAML) Configuration:
+
+```yaml
```

Comment:
So is this whole spec applicable for OAuth2-Proxy compatibility? Or are there some sections we can replace with a `...` or something to keep the sample compact and easily digestible of what the key parts are?

Which section relates to the `X-Forwarded-*` headers? Can we add a paragraph noting that Traefik sends those and that `--reverse-proxy` must be set to true for OAuth2-Proxy to consume them and be compatible with Traefik?

Reply:
Traefik always sends the X-Forwarded-* headers to the services, so it _should_ work with `ForwardAuth` and `Errors` middlewares. for the `--reverse-proxy` part: I'll write it up in a moment!